connectors/sync_job_runner.py

```diff
@@ -238,11 +240,23 @@ async def sync_starts(self):
     async def prepare_docs(self):
         logger.debug(f"Using pipeline {self.sync_job.pipeline}")
 
-        async for doc, lazy_download in self.data_provider.get_docs(
-            filtering=self.sync_job.filtering
-        ):
+        match self.sync_job.job_type:
```

Comment:
I think it would be good for the code readability if move this block of code to another method like `generator` or `create_generator(job_type)`

Reply:
Done: https://github.com/elastic/connectors-python/pull/1003/commits/d242ed9fb0263de30fab2386949ba405581912cc